dids/src/main/kotlin/web5/sdk/dids/verificationmethods/VerificationMethod.kt

```diff
@@ -0,0 +1,125 @@
+package web5.sdk.dids.verificationmethods
+
+import com.nimbusds.jose.Algorithm
+import com.nimbusds.jose.jwk.Curve
+import com.nimbusds.jose.jwk.JWK
+import web5.sdk.crypto.KeyGenOptions
+import web5.sdk.crypto.KeyManager
+import web5.sdk.dids.PublicKey
+import web5.sdk.dids.PublicKeyPurpose
+import java.util.UUID
+
+/** Common interface for options available when adding a VerificationMethod. */
+public interface VerificationMethodSpec
+
+/** Common interface for classes that can generate a [PublicKey] from a [VerificationMethodSpec]. */
+public interface VerificationMethodGenerator {
+  /**
+   * Generates a [PublicKey] from a [VerificationMethodSpec]. The first element of the pair is an optional id that can
+   * be used to identify the private key associated with the generated public key.
+   */
+  public fun generate(): Pair<String?, PublicKey>
+}
+
+/**
+ * A [VerificationMethodSpec] where a [KeyManager] will be used to generate the underlying verification method keys.
+ * The parameters [algorithm], [curve], and [options] will be forwarded to the keyManager.
+ *
+ * [relationships] will be used to determine the verification relationships in the DID Document being created.
+ * */
+public class VerificationMethodCreationParams(
+  public val algorithm: Algorithm,
+  public val curve: Curve? = null,
+  public val options: KeyGenOptions? = null,
+  public val relationships: Iterable<PublicKeyPurpose>
+) : VerificationMethodSpec {
+  internal fun toGenerator(keyManager: KeyManager): VerificationMethodKeyManagerGenerator {
+    return VerificationMethodKeyManagerGenerator(keyManager, this)
+  }
+}
+
+private const val JsonWebKey2020Type = "JsonWebKey2020"
+
+/**
+ * A [VerificationMethodSpec] according to https://w3c-ccg.github.io/lds-jws2020/.
+ *
+ * The [id] property cannot be over 50 chars and must only use characters from the Base64URL character set.
+ */
+public class JsonWebKey2020VerificationMethod(
+  public val id: String,
```

Comment:
i think this should be optional not required. default to jwk thumbprint if not provided

Reply:
Elaborating further, I don't think we should be adding for a couple of reasons. The `id` here is the verification method ID. It's not the id for the key. The key id for JWK is contained inside the `publicKeyJwk` parameter.